### DISCUSSION TITLE
How to build the project created via npm ?

### DISCUSSION THREAD

**Discussion by SaranyaPonnusamy-TDM** [directus/directus#16860] [Archive]
We are using  directus project created via npm (npm init directus-project directus)

Current docker file :

ARG NODE_VERSION=16-alpine
FROM node:${NODE_VERSION}
WORKDIR /directus
COPY . .
RUN yarn install
EXPOSE 8055
CMD ["yarn", "directus", "start"]

Instead of copying entire directory in docker file, we need to copy only the build folder for production. 
We tried running 'npm run build'. It throws below error
 'Could not find a required file.
  Name: index.js
  Searched in: /Documents/directus/src'

Note: If the project is created via npm there will be no src folder. 
How to build the project ?
I have followed the below link for setup
https://docs.directus.io/self-hosted/quickstart.html

**Comment by paescuj** [comment]
Why don't you use the [official image](https://docs.directus.io/self-hosted/installation/docker.html)?

**Reply by SaranyaPonnusamy-TDM** [reply]
sure. Thank you

**Reply by Nisthar** [reply]
why was the docs changed to docker?

**Reply by rijkvanzanten** [reply]
> why was the docs changed to docker?

Exactly because of Discussions like this 😄  The Docker image includes all dependencies and will work wherever Docker runs, which is on most hosts nowadays.
